packages/perseus/src/widgets/plotter.tsx

```diff
@@ -14,53 +13,64 @@ import KhanColors from "../util/colors";
 import GraphUtils from "../util/graph-utils";
 import KhanMath from "../util/math";
 
-import type {WidgetExports} from "../types";
+import type {APIOptions, WidgetExports} from "../types";
 
 const {deepEq} = Util;
 
-const BAR = "bar",
-    LINE = "line",
-    PIC = "pic",
-    HISTOGRAM = "histogram",
-    DOTPLOT = "dotplot";
-
-const widgetPropTypes = {
-    type: PropTypes.oneOf([BAR, LINE, PIC, HISTOGRAM, DOTPLOT]),
-    labels: PropTypes.arrayOf(PropTypes.string),
-    categories: PropTypes.arrayOf(
-        PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
-    ),
-
-    scaleY: PropTypes.number,
-    maxY: PropTypes.number,
-    snapsPerLine: PropTypes.number,
-
-    picSize: PropTypes.number,
-    pixBoxHeight: PropTypes.number,
-    picUrl: PropTypes.string,
-
-    plotDimensions: PropTypes.arrayOf(PropTypes.number),
-    labelInterval: PropTypes.number,
-    starting: PropTypes.arrayOf(PropTypes.number),
-    static: PropTypes.bool,
-} as const;
-
-class Plotter extends React.Component<any, any> {
+const BAR = "bar";
+const LINE = "line";
+const PIC = "pic";
+const HISTOGRAM = "histogram";
+const DOTPLOT = "dotplot";
+
+type Props = {
```

Comment:
We have a helper type that can provide some of the base props that each widget component accepts named: `WidgetProps<T>`. 

I'd suggest using it here, like this:

```suggestion
type Rubric = PerseusPlotterWidgetOptions;

type Props = WidgetProps<Rubric>;
```

Reply:
I couldn't make this work. Maybe we can circle back to this during a pairing?